profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -145,216 +154,283 @@ export function UntraceableState() {
+      ))}
+    </ol>
+  );
 }
 
-export function MagicNumbers(age) {
-  return (<ol>
-    { age < 18 ? <div>Spicy</div> : <div>You are not old enough</div>}
-  </ol>)
+const MIN_AGE = 18;
+export function MagicNumbers({ age }) {
+  return (
+    <ol>
+      {age >= MIN_AGE ? <div>Spicy</div> : <div>You are not old enough</div>}
+    </ol>
+  );
 }
 
 export function UnidiomaticHTMLStructure() {
-  const [name, setName] = useState("")
-  const handleSubmit = (e) => {}
+  const [name, setName] = useState("");
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
 
-  return (<div>
-    <input value={name} name="name" type="text" onChange={setName} />
-    <button type="submit" onClick={handleSubmit}>Submit</button>
-  </div>)
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        value={name}
+        name="name"
+        type="text"
+        onChange={(e) => setName(e.target.value)}
+      />
+      <button type="submit">Submit</button>
+    </form>
+  );
 }
 
 export function CrudeStateManagement() {
-  const [name, setName] = useState("")
-  const [age, setAge] = useState("")
-  const [location, setLocation] = useState("")
-  const [email, setEmail] = useState("")
-  const [password, setPassword] = useState("")
-  
-
-  const handleSubmit = (e) => {}
-
-  return (<form onSubmit={handleSubmit}>
-    <input value={name} name="name" type="text" onChange={setName} />
-    <input value={age} name="age" type="number" onChange={setAge} />
-    <input value={location} name="location" type="text" onChange={setLocation} />
-    <input value={email} name="email" type="email" onChange={setEmail} />
-    <input value={password} name="password" type="password" onChange={setPassword} />
-    <button type="submit">Submit</button>
-  </form>)
+  const [formState, setFormState] = useState({
+    name: "",
+    age: "",
+    location: "",
+    email: "",
+    password: "",
+  });
+
+  function handleFormValueChange(e) {
+    setFormState({
+      ...formState,
+      [this.name]: this.value,
+    });
+  }
+
+  const handleSubmit = (e) => {};
+  const { name, age, location, email, password } = formState;
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        value={name}
+        name="name"
+        type="text"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={age}
+        name="age"
+        type="number"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={location}
+        name="location"
+        type="text"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={email}
+        name="email"
+        type="email"
+        onChange={handleFormValueChange}
+      />
+      <input
+        value={password}
+        name="password"
+        type="password"
+        onChange={handleFormValueChange}
+      />
+      <button type="submit">Submit</button>
+    </form>
+  );
 }
 
 export function UnidiomaticHTMLHierarchy() {
```

Comment:
Don't really understand what this one is supposed to be, maybe a brief comment above can help, is this supposed to be 2 separate lists?